src/index.js

```diff
@@ -325,39 +326,51 @@ export default class Popup extends React.PureComponent {
       closeOnDocumentClick,
       className,
       on,
-      trigger,
+      trigger
     } = this.props;
-    const {modal, isOpen} = this.state;
-    const overlay = isOpen && !(on.indexOf('hover') >= 0);
+    const { modal, isOpen } = this.state;
+    const overlay = isOpen && !(on.indexOf("hover") >= 0);
     const ovStyle = modal ? styles.overlay.modal : styles.overlay.tooltip;
-    return [
-       this.renderTrigger(),
+    const portalContent = [
       isOpen && (
         <div
           key="H"
-          style={{position: 'absolute', top: '0px', left: '0px'}}
+          style={{ position: "absolute", top: "0px", left: "0px" }}
           ref={this.setHelperRef}
         />
       ),
       overlay && (
         <div
           key="O"
           className={`popup-overlay ${
-            className !== '' ? `${className}-overlay` : ''
+            className !== "" ? `${className}-overlay` : ""
           }`}
           style={Object.assign({}, ovStyle, overlayStyle)}
-          onClick={closeOnDocumentClick ? this.closePopup : undefined}>
+          onClick={closeOnDocumentClick ? this.closePopup : undefined}
+        >
           {modal && this.renderContent()}
         </div>
       ),
-      isOpen && !modal && this.renderContent(),
+      isOpen && !modal && this.renderContent()
     ];
+
+    const portalRoot = document.getElementById(this.props.popupRootID);
+    const portal = portalRoot
+      ? [
+          ReactDOM.createPortal(
```

Comment:
This is using global ReactDOM, which is unlikely to be present. Must import `ReactDOM`.